### PR TITLE
feat: upgrade to `unbuild@2`

### LINKED ISSUE #151: [feature request] support mts and cts declaration files
Context 

- https://github.com/unjs/unbuild/pull/273
- https://github.com/unjs/unbuild/pull/226

This changes are integrated in the latest version of unbuild : https://github.com/unjs/unbuild/releases/tag/v2.0.0-rc.0

Supporting them at the `module-builder` level will make it possible to support dual cjs/mjs with valid declaration types without workarounds.

The following package.json should work without a warning :  

```json
  "exports": {
    ".": {
      "require": {
        "types": "./dist/module.d.cts",
        "default": "./dist/module.cjs"
      },
      "import": {
        "types": "./dist/module.d.mts",
        "default": "./dist/module.mjs"
      },
      "types": "./dist/module.d.ts",
      "default": "./dist/module.mjs"
    }
  },
  "main": "./dist/module.cjs",
  "module": "./dist/module.mjs",
  "types": "./dist/types.d.ts",
  "files": [
    "dist",
    "*.d.ts",
    "*.cjs",
    "*.mjs"
  ],
```

### PR DESCRIPTION
**Pull request by antfu**
Resolves https://github.com/nuxt/module-builder/issues/151

### CODE REVIEW

File: src/build.ts [hunk 1]
```diff
@@ -81,14 +82,21 @@ export async function buildModule (opts: BuildModuleOptions) {
...
   if (existsSync(dtsFile)) {
     return
   }
 
   // Read generated module types
   const moduleTypesFile = resolve(distDir, 'module.d.ts')
   const moduleTypes = await fsp.readFile(moduleTypesFile, 'utf8').catch(() => '')
-  const typeExports = findExports(moduleTypes)
+  const typeExports = findExports(
+    // Replace `export { type Foo }` with `export { Foo }`
```
Comment (antfu) [inline src/build.ts:94]: This is due to https://github.com/Swatinem/rollup-plugin-dts/commit/1b228d052785ea7ad061967430d3ed560582c46c

I am uncertain if such a feature should be included in `mlly` as `findExports` is used to find non-type exports. While `fineTypeExports` finds types exports only. But in `.d.ts` actually, all exports are type exports with or without the `type` indicator.

**Comment by Hebilicious** [comment]
Resolves #151

**Review by danielroe — APPROVED** [review]

File: test/build.spec.ts [hunk 2]
```diff
@@ -19,11 +19,14 @@ describe('module builder', () => {
     expect(files).toMatchInlineSnapshot(`
       [
         "module.cjs",
+        "module.d.mts",
```
Comment (Hebilicious) [inline test/build.spec.ts:22]: It would be great if out the box we would generate mts and cts 🙏🏽

Comment (danielroe) [reply test/build.spec.ts:22]: what do you mean?

**Pull request merged**